src/wallet/salvage.cpp

```diff
@@ -16,14 +16,14 @@ static const char *HEADER_END = "HEADER=END";
 static const char *DATA_END = "DATA=END";
 typedef std::pair<std::vector<unsigned char>, std::vector<unsigned char> > KeyValPair;
 
-bool RecoverDatabaseFile(const fs::path& file_path)
+bool RecoverDatabaseFile(const fs::path& file_path, bilingual_str& error, std::vector<bilingual_str>& warnings)
 {
     std::string filename;
     std::shared_ptr<BerkeleyEnvironment> env = GetWalletEnv(file_path, filename);
 
     bilingual_str open_err;
     if (!env->Open(open_err)) {
-        tfm::format(std::cerr, "%s\n", open_err.original);
+        error = strprintf(Untranslated("Error initializing wallet database environment %s!"), env->Directory());
```

Comment:
in commit b27b5f7307:

why is open_error ignored and overwritten?

Reply:
Dunno. I've restored the previous behavior.